### PR TITLE
ARCH-1536 - API Success Review - Report Comments (POST, PUT and DELETE v4)

### PR DESCRIPTION
**Pull request by ArunavaSAP**

### CODE REVIEW

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 1]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
     }
 ]
 ```
+## <a name="post-report-header-comments"></a>Post Report Header Comments
+
+Creates a new comment for the specified report.
+
+### Scopes
+
+`expense.report.read` - Refer to [Scope Usage](#scope-usage) for full details.
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:244]: expense.report.readwrite scope is used for POST API. Please refer https://github.concur.com/inception/expense-report-service/blob/develop/src/main/java/com/concur/expense/web/controllers/CommentController.java#L140

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 2]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+* [RFC 7232 If-Modified-Since](https://tools.ietf.org/html/rfc7232#section-3.3)
+* [RFC 7231 Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4)
+* [RFC 7235 Authorization](https://tools.ietf.org/html/rfc7235#section-4.2) - Bearer Token that identifies the caller. This is the Company or User access token.
+* `concur-correlationid` is a Concur specific custom header used for technical support in the form of a [RFC 4122 A Universally Unique IDentifier (UUID) URN Namespace](https://tools.ietf.org/html/rfc4122)
+
+### Response
+
+#### Status Codes
+
+* [200 OK](https://tools.ietf.org/html/rfc7231#section-6.3.2)
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:277]: https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1 refers to 200 OK. Please check other links as well.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 3]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+## <a name="post-report-header-comments"></a>Post Report Header Comments
+
+Creates a new comment for the specified report.
+
+### Scopes
+
+`expense.report.read` - Refer to [Scope Usage](#scope-usage) for full details.
+
+### Request
+
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:247]: Don't we require to add the request body for POST?

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 4]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+}
+```
+
+## <a name="Update-report-header-comments"></a>Update Report Header Comments
+
+Updates the latest comment for the specified report.
+
+### Scopes
+
+`expense.report.read` - Refer to [Scope Usage](#scope-usage) for full details.
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:324]: expense.report.readwrite scope is used for PATCH API. Please refer https://github.concur.com/inception/expense-report-service/blob/develop/src/main/java/com/concur/expense/web/controllers/CommentController.java#L172

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 5]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+
+### Request
+
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:327]: Don't we require to add the request body for PATCH? Please refer other files which has POST and PATCH APIs.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 6]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+```
+
+#### Parameters
+
+|Name|Type|Format|Description|
+|---|---|---|---|
+|`userID`|`string`|-|**Required** The unique identifier of the SAP Concur user. Use [Identity v4](/api-reference/profile/v4.identity.html) to retrieve the `userID`.|
+|`contextType`|`string`|-|**Required** The access level of the SAP Concur user, which determines the form fields they can view/modify. Supported values: `TRAVELER`, `PROXY`|
+|`reportId`|`string`|-|**Required** The unique identifier of the report that is being read.|
+
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:261]: We have a optional parameter `accept-language`.  Please check that we should add that too or not.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 7]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:341]: We have a optional parameter `accept-language`.  Please check that we should add that too or not.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 8]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+* [RFC 7232 If-Modified-Since](https://tools.ietf.org/html/rfc7232#section-3.3)
+* [RFC 7231 Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4)
+* [RFC 7235 Authorization](https://tools.ietf.org/html/rfc7235#section-4.2) - Bearer Token that identifies the caller. This is the Company or User access token.
+* `concur-correlationid` is a Concur specific custom header used for technical support in the form of a [RFC 4122 A Universally Unique IDentifier (UUID) URN Namespace](https://tools.ietf.org/html/rfc4122)
+
+### Response
+
+#### Status Codes
+
+* [200 OK](https://tools.ietf.org/html/rfc7231#section-6.3.2)
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:357]: https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1 refers to 200 OK. Please check other links as well.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 9]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+}
+```
+
+## <a name="Delete-report-header-comments"></a>Delete Report Header Comments
+
+Deletes the latest comment for the specified report.
+
+### Scopes
+
+`expense.report.read` - Refer to [Scope Usage](#scope-usage) for full details.
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:405]: expense.report.delete scope is used for DELETE API. Please refer
https://github.concur.com/inception/expense-report-service/blob/develop/src/main/java/com/concur/expense/web/controllers/CommentController.java#L206

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 10]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+```
+
+#### Parameters
+
+|Name|Type|Format|Description|
+|---|---|---|---|
+|`userID`|`string`|-|**Required** The unique identifier of the SAP Concur user. Use [Identity v4](/api-reference/profile/v4.identity.html) to retrieve the `userID`.|
+|`contextType`|`string`|-|**Required** The access level of the SAP Concur user, which determines the form fields they can view/modify. Supported values: `TRAVELER`, `PROXY`|
+|`reportId`|`string`|-|**Required** The unique identifier of the report that is being read.|
+
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:422]: We have a optional parameter accept-language. Please check that we should add that too or not.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 11]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+* [RFC 7232 If-Modified-Since](https://tools.ietf.org/html/rfc7232#section-3.3)
+* [RFC 7231 Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4)
+* [RFC 7235 Authorization](https://tools.ietf.org/html/rfc7235#section-4.2) - Bearer Token that identifies the caller. This is the Company or User access token.
+* `concur-correlationid` is a Concur specific custom header used for technical support in the form of a [RFC 4122 A Universally Unique IDentifier (UUID) URN Namespace](https://tools.ietf.org/html/rfc4122)
+
+### Response
+
+#### Status Codes
+
+* [204 No Content](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.5)
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:438]: https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1 refers to 200 OK. Please check other links as well.

**Review by GeethaMaiKK — CHANGES_REQUESTED** [review]

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 12]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
+* [500 Internal Server Error](https://tools.ietf.org/html/rfc7231#section-6.6.1)
+
+
+#### Headers
+
+* `concur-correlationid` is a Concur specific custom header used for technical support in the form of a [RFC 4122 A Universally Unique IDentifier (UUID) URN Namespace](https://tools.ietf.org/html/rfc4122)
+
+#### Payload
+
+* [Report Header Comments Response](#report-header-comments-response-schema)
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:291]: This is not the response for the POST API. This is the response for GET API. I think you have to create a new table for POST response with uri.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 13]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:371]: This is not the response for the PATCH API. This is the response for GET API. I think you have to create a new table for PATCH response with uri.

File: src/api-reference/expense/expense-report/v4.comments.markdown [hunk 14]
```diff
@@ -235,6 +235,240 @@ curl --location --request GET 'https://us.api.concursolutions.com/expensereports
...
```
Comment (GeethaMaiKK) [inline src/api-reference/expense/expense-report/v4.comments.markdown:452]: We don't have any response payload for DELETE API. Please check other files which has same scenario.

**Review by GeethaMaiKK — CHANGES_REQUESTED** [review]

**Review by RoAnnC1 — APPROVED** [review]
:shipit:

**Pull request merged**